forem/forem
# Refactor: FB Auth provider using SDK instead of OAuth flow

## What type of PR is this? (check all applicable)


## Description

At the moment this is still a WIP

To-Do:
- [x] Register/Login
- [x] Connect account (from `/settings`)

## Related Tickets & Documents

Closes #14681

## QA Instructions, Screenshots, Recordings

1. Enable FB Auth `/admin/customization/config` as admin
2. Use Incognito window and register a new account using FB
   - It should take you to onboarding
3. Delete the user from DB (either console `.destroy` or `/admin/users/<id>/edit`)
4. Log in with any user (could be `admin_mcadmin`) and go to the settings page
5. Connect the existing account with FB
   - It should work and the connect with FB option should appear in settings anymore
6. Log out and then log in with facebook to that account

### UI accessibility concerns?

None: The buttons weren't modified so I don't think accessibility would be affected.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post

## [optional] What gif best describes this PR or how it makes you feel?

![keylock internals](https://media.giphy.com/media/QvSGhHq8CrVzq/giphy.gif?cid=ecf05e47139b03w30ay2v173l1o6iciy8u2h3kgyq4cuzmtb&rid=giphy.gif&ct=g)
